rom/src/rom.rs

```diff
@@ -236,7 +236,21 @@ impl Soc {
         self.registers
             .fuse_soc_stepping_id
             .write(soc::bits::FuseSocSteppingId::SocSteppingId.val(soc_stepping_id));
-        // TODO: debug unlock / rma token?
+
+        if fuses.cptra_ss_manuf_debug_unlock_token().len()
+            != self.registers.fuse_manuf_dbg_unlock_token.len() * 4
+        {
+            romtime::println!("[mcu-fuse-write] SS manuf debug unlock token length mismatch");
+            fatal_error(1);
+        }
+        for i in 0..fuses.cptra_ss_manuf_debug_unlock_token().len() / 4 {
+            let word = u32::from_le_bytes(
+                fuses.cptra_ss_manuf_debug_unlock_token()[i * 4..i * 4 + 4]
+                    .try_into()
+                    .unwrap(),
+            );
+            self.registers.fuse_manuf_dbg_unlock_token[i].set(word);
+        }
```

Comment:
nit: What do you think of using an iterator here instead?


```suggestion
        for (reg, fuses) in core::iter::zip(self.registers.fuse_manuf_dbg_unlock_token, fuses.cptra_ss_manuf_debug_unlock_token().chunks_exact(4) {
            let word = u32::from_le_bytes(fuses.try_into().unwrap());
            reg.set(word);
        }
```

Reply:
This causes an illegal move because self.registers.fuse_manuf_dbg_unlock_token does not implement the Copy trait. So I think we are best using slices here.

Reply:
Okay, this works for me!